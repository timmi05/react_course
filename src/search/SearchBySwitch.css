.search-by-switch{
    display: flex;
}

.search-by-switch-label{
    padding-right: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}

.search-by-switch-button{
    font-size: 15px;
    padding: 10px 26px;
}