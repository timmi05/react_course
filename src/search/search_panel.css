.search-panel {
    margin: 40px;
}

.search-panel-label{
    margin-bottom: 30px;
    font-size: 40px;
}

.search-panel-main{
    padding-bottom: 15px;
    display: flex;
}

.search-panel-input{
    width: 85%;
    margin-right: 20px;
    height: 40px;
    font-size: 28px;
}

.search-panel-button{
    font-size: 20px;
    padding: 10px 35px;
}