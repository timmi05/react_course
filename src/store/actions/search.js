export const search = (value, searchBy) => ({
    type: 'SEARCH_MOVIES',
    payload: {
        value: value,
        searchBy: searchBy
    }
});

export const moveToSearch = () => ({
    type: 'MOVE_TO_SEARCH'
});
