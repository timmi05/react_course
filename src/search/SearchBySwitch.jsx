import React, { useState } from 'react';
import { Button } from '../component/Button'
import './SearchBySwitch.css';

export const SearchBySwitch = () => {

    const [disabled, setDisabled] = useState(false);

    const SearchByTitle = () => {
        setDisabled(true);
    };

    const SearchByGenre = () => {
        setDisabled(false);
    };

    return (
        <div className='search-by-switch'>
            <span className='search-by-switch-label'>SEARCH BY </span>
            <div className = 'search-by-switch-buttons'>
                <Button
                label = 'TITLE'
                classes = 'search-by-switch-button search-by-switch-left-button'
                onClick = { SearchByTitle }
                disabled = { disabled }
                />
                <Button
                label = 'GENRE'
                classes = 'search-by-switch-button search-by-switch-right-button'
                onClick = { SearchByGenre }
                disabled = { !disabled }
                />
            </div>
        </div>
    );
};
