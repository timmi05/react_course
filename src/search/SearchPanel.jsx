import React, { useState }  from 'react';
import { SearchBySwitch } from './SearchBySwitch';
import { Button } from '../component/Button'
import './SearchPanel.css';

export const SearchPanel = () => {

    const [value, setValue] = useState('');
    const [disabled, setDisabled] = useState(false);

    const handleChange = event => {
      setValue(event.target.value);
    };

    const handleSubmit = () => {
      setDisabled(true);
    };

    return (
        <div className='search-panel'>
           <div className='search-panel-label'>FINED YOUR MOVIE</div>
           <div className='search-panel-main'>
               <input className='search-panel-input' type='text' value={ value } onChange={ handleChange } />
               <Button
               label = 'SEARCH'
               classes = 'search-panel-button'
               onClick = { handleSubmit }
               disabled = { disabled }
               />
           </div>
           <SearchBySwitch/>
        </div>
    );
};
