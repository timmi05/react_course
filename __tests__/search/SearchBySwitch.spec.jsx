import React from 'react';
import { mount } from 'enzyme';
import { SearchBySwitch } from '../../src/search/SearchBySwitch';

describe('SearchBySwitch rendering', () => {

    it('renders correctly', () => {
        const component = mount(<SearchBySwitch/>);
        expect(component).toMatchSnapshot();
    });

    it('simulate click on button by TITLE', () => {
        const component = mount(<SearchBySwitch/>);
        component.find({label: 'TITLE'}).simulate('click');
        expect(component.find({className:'button search-by-switch-button search-by-switch-left-button'})
            .prop('disabled')).toBe(true);
        expect(component).toMatchSnapshot();
    });

    it('simulate click on button by GENRE', () => {
        const component = mount(<SearchBySwitch/>);
        component.find({label: 'TITLE'}).simulate('click');
        component.find({label: 'GENRE'}).simulate('click');
        expect(component.find({className:'button search-by-switch-button search-by-switch-right-button'})
            .prop('disabled')).toBe(true);
        expect(component).toMatchSnapshot();
    });
});
