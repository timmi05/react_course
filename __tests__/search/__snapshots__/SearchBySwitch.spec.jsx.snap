// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBySwitch rendering renders correctly 1`] = `
<SearchBySwitch>
  <div
    className="search-by-switch"
  >
    <span
      className="search-by-switch-label"
    >
      SEARCH BY 
    </span>
    <div
      className="search-by-switch-buttons"
    >
      <Button
        classes="search-by-switch-button search-by-switch-left-button"
        disabled={false}
        label="TITLE"
        onClick={[Function]}
      >
        <button
          className="button search-by-switch-button search-by-switch-left-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          TITLE
        </button>
      </Button>
      <Button
        classes="search-by-switch-button search-by-switch-right-button"
        disabled={true}
        label="GENRE"
        onClick={[Function]}
      >
        <button
          className="button search-by-switch-button search-by-switch-right-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          GENRE
        </button>
      </Button>
    </div>
  </div>
</SearchBySwitch>
`;

exports[`SearchBySwitch rendering simulate click on button by GENRE 1`] = `
<SearchBySwitch>
  <div
    className="search-by-switch"
  >
    <span
      className="search-by-switch-label"
    >
      SEARCH BY 
    </span>
    <div
      className="search-by-switch-buttons"
    >
      <Button
        classes="search-by-switch-button search-by-switch-left-button"
        disabled={false}
        label="TITLE"
        onClick={[Function]}
      >
        <button
          className="button search-by-switch-button search-by-switch-left-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          TITLE
        </button>
      </Button>
      <Button
        classes="search-by-switch-button search-by-switch-right-button"
        disabled={true}
        label="GENRE"
        onClick={[Function]}
      >
        <button
          className="button search-by-switch-button search-by-switch-right-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          GENRE
        </button>
      </Button>
    </div>
  </div>
</SearchBySwitch>
`;

exports[`SearchBySwitch rendering simulate click on button by TITLE 1`] = `
<SearchBySwitch>
  <div
    className="search-by-switch"
  >
    <span
      className="search-by-switch-label"
    >
      SEARCH BY 
    </span>
    <div
      className="search-by-switch-buttons"
    >
      <Button
        classes="search-by-switch-button search-by-switch-left-button"
        disabled={true}
        label="TITLE"
        onClick={[Function]}
      >
        <button
          className="button search-by-switch-button search-by-switch-left-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          TITLE
        </button>
      </Button>
      <Button
        classes="search-by-switch-button search-by-switch-right-button"
        disabled={false}
        label="GENRE"
        onClick={[Function]}
      >
        <button
          className="button search-by-switch-button search-by-switch-right-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          GENRE
        </button>
      </Button>
    </div>
  </div>
</SearchBySwitch>
`;
